analyses/upstream-analysis/04_run_filter_object.Rmd

```diff
@@ -0,0 +1,381 @@
+
+```{r filter-ambient-RNA, echo=TRUE}
+# Indicates whether or not to use the SoupX corrected matrix for filtering or to skip it
+if (use_SoupX_filtering == "YES"){
+  print_message <- print("we will use SoupX corrected matrix")
+  SoupX_seurat_obj <- readRDS(SoupX_file) # Read SoupX_seurat_obj
+  colnames(seurat_obj)[which(!(colnames(seurat_obj) %in% colnames(SoupX_seurat_obj) ))]
+  seurat_obj <- subset(seurat_obj, cells=colnames(seurat_obj)[which(!(colnames(seurat_obj) %in% colnames(SoupX_seurat_obj) ))], invert=TRUE)
+  SoupX_seurat_obj <- SoupX_seurat_obj[,colnames(seurat_obj)]
+  seurat_obj[["RNA_SoupX"]] <- CreateAssayObject(counts = SoupX_seurat_obj)
+  DefaultAssay(seurat_obj) <- "RNA_SoupX"
+  cat("Cells_number_after_filter_ambient_RNA per orig.ident", "\n")
+  print(table(seurat_obj@meta.data$orig.ident))
+  cat("Cells_number_after_filter_ambient_RNA per ID", "\n")
+  print(table(seurat_obj@meta.data$ID))
+  } else {
+    print_message <- print("we will skip usage of the SoupX corrected matrix")}
+```
+
+Here, `r print_message`.
+
+# Filter doublets out
+Next, we will remove any doublets from all samples as identified by [ScDblFinder](https://bioconductor.org/packages/devel/bioc/vignettes/scDblFinder/inst/doc/scDblFinder.html).
+
+```{r filter-doublets, echo=TRUE}
+# Read scDblFinder results
+scDblFinder_obj <- readRDS(scDblFinder_file)
+
+#############################################################################
+# Estimate `cells_number_raw` per sample_name
+cells_number_raw <- table(scDblFinder_obj@meta.data$orig.ident) %>%
+    as.data.frame() %>% 
+    mutate(sample_name = Var1,
+           cells_number_raw = Freq) %>% 
+    dplyr::select(-Var1, -Freq)
+
+#############################################################################
+# Add metadata
+seurat_obj <- AddMetaData(seurat_obj, scDblFinder_obj@meta.data)
+
+##########################################################################################################
+# Report Singlets 
+singlet_cells <- colnames(scDblFinder_obj)[which(scDblFinder_obj@meta.data$scDblFinder.class == "singlet")]
+seurat_obj_doublets <- subset(seurat_obj, cells = singlet_cells, invert = TRUE)
+
+cells_number_doublets <- table(seurat_obj_doublets@meta.data$orig.ident) %>%
+    as.data.frame() %>% 
+    mutate(sample_name = Var1,
+           cells_number_doublets = Freq) %>% 
+    dplyr::select(-Var1, -Freq)
+##########################################################################################################
+# Report Doublets
+doublet_cells <- colnames(seurat_obj)[which(seurat_obj@meta.data$scDblFinder.class == "doublet")]
+
+##########################################################################################################
+# Filter Doublets out
+# Unique cells to keep
```

Comment:
Again, very minor, but possibly for clarity, I feel like maybe it would be more clear to comment this as unique cells to remove?

Reply:
```suggestion
# Unique cells to filter
```

Reply:
Addressed here: 3e21b97aff92cc9d23005375f62c440bbedfb01f